src/main/java/graphql/Scalars.java

```diff
@@ -69,7 +69,7 @@ private Integer convertImpl(Object input) {
         public Integer serialize(Object input) {
             Integer result = convertImpl(input);
             if (result == null) {
-                throw new CoercingSerializeException("Invalid value '" + input + "' for Int");
+                throw new CoercingSerializeException("Expected type 'Int'.");
```

Comment:
My problem here is that its correct but perhaps too terse.

The system is yelling "I want a Int" but its not giving any clues into what you gave. (say an Boolean accidently)

Perhaps something like

    throw new CoercingSerializeException("Expected type 'Int' but received a '" + mkActual(input) "'.");

   String mkActual(Object input) {
        if (input == null) {
              return "null";
        } else {
              return input.getClass().getSimpleName();
       }
  }


Yes it leaks the Java idiom but it will tell you more when the bug reports starting coming in.

Dev A :  It says it wants an Int
Dev B:  And what did you give it?
Dev A: Umm...oh it says Boolean
Dev B: well there you go.  Fix that!